describeconfigs.go

```diff
@@ -0,0 +1,305 @@
+}
+
+func (t *describeConfigsResponseV0) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readInt32(r, size, &t.ThrottleTimeMs); err != nil {
+		return
+	}
+	fn := func(r *bufio.Reader, size int) (fnRemain int, fnErr error) {
+		var resource describeConfigsResponseV0Resource
+		if fnRemain, fnErr = (&resource).readFrom(r, size); err != nil {
+			return
+		}
+		t.Resources = append(t.Resources, resource)
+		return
+	}
+	if remain, err = readArrayWith(r, remain, fn); err != nil {
+		return
+	}
+
+	return
+}
+
+func (c *Conn) describeConfigs(request describeConfigsRequestV0) (describeConfigsResponseV0, error) {
+	var response describeConfigsResponseV0
+	err := c.writeOperation(
+		func(deadline time.Time, id int32) error {
+			return c.writeRequest(describeConfigs, v0, id, request)
+		},
+		func(deadline time.Time, size int) error {
+			return expectZeroSize(func() (remain int, err error) {
+				return (&response).readFrom(&c.rbuf, size)
+			}())
+		},
+	)
+	if err != nil {
+		return response, err
+	}
+	for _, tr := range response.Resources {
+		if tr.ErrorCode != 0 {
+			return response, Error(tr.ErrorCode)
+		}
+	}
+	return response, nil
+}
+
+type ResourceType int8
+
+const (
+	ResourceTypeUnknown ResourceType = 0
+	ResourceTypeAny     ResourceType = 1
+	ResourceTypeTopic   ResourceType = 2
+	ResourceTypeGroup   ResourceType = 3
+	ResourceTypeCluster ResourceType = 4
+	ResourceTypeBroker  ResourceType = 5
+)
+
+//DescribeConfig
+type DescribeConfig struct {
+	//ResourceType
+	//0: Unknown
+	//1: Any
+	//2: Topic
+	//3: Group
+	//4: Cluster
+	//5: Broker (new)
+	ResourceType ResourceType
+
+	ResourceName string
+
+	ConfigNames []string
+}
+
+//DescribeConfigsResponse
+type DescribeConfigsResponse struct {
+	ThrottleTimeMs int32
+	Resources      []DescribeConfigsResponseResource
+}
+
+//DescribeConfigsResponseResource
+type DescribeConfigsResponseResource struct {
+	ErrorCode     int16
+	ErrorMessage  string
+	ResourceType  int8
+	ResourceName  string
+	ConfigEntries []DescribeConfigsResponseConfigEntry
+}
+
+//DescribeConfigsResponseConfigEntry
+type DescribeConfigsResponseConfigEntry struct {
+	ConfigName  string
+	ConfigValue string
+	ReadOnly    bool
+	IsDefault   bool
+	IsSensitive bool
+}
+
+//DescribeConfigs gets configuration data.
+//
+//Detailed info please look at https://cwiki.apache.org/confluence/display/KAFKA/KIP-133%3A+Describe+and+Alter+Configs+Admin+APIs
+func (c *Conn) DescribeConfigs(configs ...DescribeConfig) (DescribeConfigsResponse, error) {
+	var requestV0Resource []describeConfigsRequestV0Resource
+	for _, t := range configs {
+		requestV0Resource = append(
+			requestV0Resource,
+			describeConfigsRequestV0Resource{
+				ResourceType: int8(t.ResourceType),
+				ResourceName: t.ResourceName,
+				ConfigNames:  t.ConfigNames,
+			})
+	}
+
+	resV0, err := c.describeConfigs(describeConfigsRequestV0{
+		Resources: requestV0Resource,
+	})
+
+	res := convertToPublicAPI(resV0)
```

Comment:
we should handle the error before trying to convert the response `resV0`

Reply:
actually, the *describeConfigs* function always return response even if it has some error. so here we didn't need to judge error first.